### PR TITLE
fix: project link not set in accounts other than profilt and loss acc

### PR DESCRIPTION
**Pull request by nextchamp-saqib**
- Item wise projects are prioritised : Setting project in Sales Invoice Item will not be overwritten by Sales Invoice project.
- Projects & Cost Centres can be linked to all types of accounts while making GL Entry/Journal Entry

### CODE REVIEW

File: erpnext/controllers/stock_controller.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def get_gl_entries(self, warehouse_account=None, default_expense_account=None,
 							"account": warehouse_account[sle.warehouse]["account"],
 							"against": item_row.expense_account,
 							"cost_center": item_row.cost_center,
+							"project": item_row.project or self.project if hasattr(self, 'project') else None,
```
Comment (deepeshgarg007) [inline erpnext/controllers/stock_controller.py:95]:
```suggestion
							"project": item_row.project or self.project if hasattr(self, 'project'),
```

File: erpnext/controllers/stock_controller.py [hunk 2]
```diff
@@ -101,6 +102,7 @@ def get_gl_entries(self, warehouse_account=None, default_expense_account=None,
 							"account": item_row.expense_account,
 							"against": warehouse_account[sle.warehouse]["account"],
 							"cost_center": item_row.cost_center,
+							"project": item_row.project or self.project if hasattr(self, 'project') else None,
```
Comment (deepeshgarg007) [inline erpnext/controllers/stock_controller.py:106]:
```suggestion
							"project": item_row.project or self.project if hasattr(self, 'project'),
```

**Review by deepeshgarg007 — COMMENTED** [review]
@nextchamp-saqib Test cases are also failing, please fix them

**Review by nabinhait — COMMENTED** [review]
@nextchamp-saqib Please fix the conflicts

**Comment by nextchamp-saqib** [comment]
This will be taken up by @deepeshgarg007. He might open up a new PR altogether.

**Comment by justlevine** [comment]
whats the status on this?

**Pull request closed without merging**